### PR TITLE
Refactor Grid components to function components

### PR DESCRIPTION
**Pull request by ethanliu24**
## Proposed Changes

_(Describe your changes here. Also describe the motivation for your changes: what problem do they solve, or how do they improve the application or codebase? If this pull request fixes an open issue, [use a keyword to link this pull request to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).)_

Following React's recommendation on using function components, this PR refactors the components in `/grid` folder to function components from class components.
...

<details>
<summary>Screenshots of your changes (if applicable)</summary>

</details>

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |          |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |X         |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

- [X] I have performed a self-review of my changes.
  - Check that all changed files included in this pull request are intentional changes.
  - Check that all changes are relevant to the purpose of this pull request, as described above.
- [ ] I have added tests for my changes, if applicable.
  - This is **required** for all bug fixes and new features.
- [ ] I have updated the project documentation, if applicable.
  - This is **required** for new features.
- [ ] If this is my first contribution, I have added myself to the list of contributors.
- [x] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

- [x] I have verified that the CircleCI checks have passed.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

### CODE REVIEW

**Comment by david-yz-liu** [comment]
@ethanliu24 I'll review this tomorrow, but you should really provide a more descriptive pull request title

File: js/components/grid/calendar.js.jsx [hunk 1]
```diff
@@ -1,301 +1,285 @@
...
+  // (go to Lecture constructor for definition of 'width')
+  setWidths(fallSession)
+  setWidths(springSession)
+
+  // Fill session colSpans dictionaries
+  const fallColSpans = { M: 0, T: 0, W: 0, R: 0, F: 0 }
+  const springColSpans = { M: 0, T: 0, W: 0, R: 0, F: 0 }
+  storeColSpans(fallSession, fallColSpans)
+  storeColSpans(springSession, springColSpans)
+
```
Comment (david-yz-liu) [inline js/components/grid/calendar.js.jsx:37]: Revert this change

File: js/components/grid/calendar.js.jsx [hunk 2]
```diff
@@ -1,301 +1,285 @@
...
-        <TimetableContainer
-          session="S"
-          lecturesByTime={springSession}
-          headColSpans={springColSpans}
-          lectures={springLectures}
-        />
-      </>
-    )
-  }
+export const Row = props => {
```
Comment (david-yz-liu) [inline js/components/grid/calendar.js.jsx:8]: You should be able to use the standard function syntax here and throughout, `export function Row(props) { ... }`

File: js/components/grid/calendar.js.jsx [hunk 3]
```diff
@@ -1,301 +1,285 @@
...
-    this.displayMap = this.displayMap.bind(this)
-  }
-
-  displayMap(session) {
-    this.modal.current.openModal()
-  }
+const Timetable = props => {
+  const modal = createRef()
+  const displayMap = session =>
+    useCallback(() => {
```
Comment (david-yz-liu) [inline js/components/grid/calendar.js.jsx:80]: The documentation for `useCallback` indicates that it should be defined at the top-level of the component. This suggests you should define a first constant `const openMapModal = useCallback(...)` and then use that to define `displayMap`.

File: js/components/grid/calendar.js.jsx [hunk 4]
```diff
@@ -1,301 +1,285 @@
...
+  } else {
+    // Adjust the order at which these cells are rendered
+    if (props.session === "F") {
+      tableData.unshift(dummyCell)
+    } else if (props.session === "S") {
+      tableData.push(dummyCell)
     }
-    return <tr>{tableData}</tr>
   }
+
```
Comment (david-yz-liu) [inline js/components/grid/calendar.js.jsx:281]: revert this change

**Review by david-yz-liu — COMMENTED** [review]
@ethanliu24 great job! I left a few comments. Please go ahead and do the conversion for the `course_panel.js.jsx` file as well!

File: js/components/grid/calendar.js.jsx (hunk 3 above)
Comment (ethanliu24) [reply js/components/grid/calendar.js.jsx:80]: Should I keep the `session` variable in there, which is not used?
